jobs/garbage-collection.Jenkinsfile

```diff
@@ -0,0 +1,91 @@
+        ])
+    ])
+
+    def build_description = "[${params.STREAM}] Garbage Collection"
+    def cosa_img = pipeutils.get_cosa_img(pipecfg, params.STREAM)
+    // Grab any environment variables we should set
+    def container_env = pipeutils.get_env_vars_for_stream(pipecfg, params.STREAM)
+    // Keep in sync with build.Jenkinsfile
+    def cosa_memory_request_mb = 10.5 * 1024 as Integer
+    def ncpus = ((cosa_memory_request_mb - 512) / 1536) as Integer
+    def timeout_mins = 240
+    def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, params.STREAM)
+
+    lock(resource: "gc-${params.STREAM}") {
+        // Read YAML file
+        def gc_policy_data = readYaml file: './gc-policy.yaml'
+        
+        cosaPod(image: cosa_img, env: container_env,
+                cpu: "${ncpus}", memory: "${cosa_memory_request_mb}Mi",
+                serviceAccount: "jenkins") {
+            timeout(time: timeout_mins, unit: 'MINUTES') {
+
+                try {
+                    // Write YAML data to a new file in cosaPod
+                    def new_gc_policy_path = './gc-policy.yaml'
+                    writeYaml file: new_gc_policy_path, data: gc_policy_data
+
+                    currentBuild.description = "${build_description} Running"
+
+                    def credentials = [file(variable: 'GCP_KOLA_TESTS_CONFIG', credentialsId: 'gcp-image-upload-config')]
+
+                    withCredentials(credentials) {
+                        // pick up the GCP project to use from the config
+                        def gcp_project = sh(script: "jq -r .project_id \${GCP_KOLA_TESTS_CONFIG}", returnStdout: true).trim()
+                        def acl = pipecfg.s3.acl ?: 'public-read'
+                        def dry_run = params.DRY_RUN ? "--dry-run" : ""
+                        stage('Init') {
+                            def branch = params.STREAM
+                            shwrap("cosa init --force --branch ${branch} https://github.com/${repo}")
+                        }
+                        stage("Buildfetch") {
+                            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                            cosa buildfetch --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG} \
+                                --stream ${params.STREAM} --url=s3://${s3_stream_dir}/builds
+                            """)
+                        }
+                        stage('Garbage Collection') {
+                            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                            cosa cloud-prune --policy ${new_gc_policy_path} \
+                            --stream ${params.STREAM} ${dry_run} \
+                            --gcp-project=${gcp_project} \
+                            --gcp-json-key=\${GCP_KOLA_TESTS_CONFIG} \
+                            --acl=${acl} \
+                            --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG}
+                            """)
```

Comment:
We have handled this situation pretty graciously in the `cloud-prune` script i.e. If the artifacts are already/recently deleted the deletion process would return SUCCESS.
So ideally, if we delete some resources, and don't end up uploading the new `builds.json`, it will take care of it in the next rerun.